spec/features/admin/bulk_product_update_spec.rb

```diff
@@ -514,37 +514,61 @@
         login_as_admin_and_visit spree.admin_products_path
       end
 
-      it "shows an edit button for products, which takes the user to the standard edit page for that product in a new window" do
+      it "shows an edit button for products, which takes the user to the standard edit page for that product" do
         expect(page).to have_selector "a.edit-product", count: 2
 
-        new_window = window_opened_by do
-          within "tr#p_#{p1.id}" do
-            find("a.edit-product").click
-          end
+        within "tr#p_#{p1.id}" do
+          find("a.edit-product").click
         end
 
-        within_window new_window do
-          expect(URI.parse(current_url).path).to eq "/admin/products/#{p1.permalink}/edit"
-          page.execute_script('window.close()')
+        expect(URI.parse(current_url).path).to eq spree.edit_admin_product_path(v1.product.permalink)
+      end
+
+      it "shows an edit button for products, which takes the user to the standard edit page for that product, url includes selected filter" do
+        expect(page).to have_selector "a.edit-product", count: 2
+
+        # Set a filter
+        select2_select p1.supplier.name, from: "producer_filter"
+        apply_filters
+
+        within "tr#p_#{p1.id}" do
+          find("a.edit-product").click
         end
+
+        uri = URI.parse(current_url)
+        expect("#{uri.path}?#{uri.query}").to eq spree.edit_admin_product_path(v1.product.permalink, producerFilter: p1.supplier.id)
```

Comment:
isn't this the same as `expect(page).to have_current_path(spree.edit_admin_product_path(v1.product.permalink, producerFilter: p1.supplier.id))`?